### PR TITLE
Value card support in dashboard editor

### PR DESCRIPTION
**Pull request by JoelArmendariz**
Issue: https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1740

**Summary**

Support the use of ValueCards in the DashboardEditor

**Change List (commits, features, bugs, etc)**

 - Adds:
`DataSeriesFormItemModal`
`ThresholdsFormItem`
`SimpleIconDropdown`

 - Updates various utility functions
 - Adds styling for new modal

**Acceptance Test (how to verify the PR)**

Go to the DashboardEditor default story and add a ValueCard.
Make sure that all of the functionality updates the JSON properly as defined here: https://www.ibm.com/support/knowledgecenter/en/SSQP8H/iot/dashboard/value_json_ref.html
Make sure styling is correct and that there are no edge cases with configurations.

### CODE REVIEW

File: src/components/CardEditor/CardEditForm/CardEditFormItems/ThresholdsFormItem.jsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                  const thresholdIndex = thresholds.findIndex(
+                    (item) => item.id === threshold.id
+                  );
+                  const filteredThresholds = thresholds.filter(
+                    (item, index) => index !== thresholdIndex
+                  );
+                  onChange(filteredThresholds.map((item) => omit(item, 'id')));
+                  setThresholds(filteredThresholds);
+                }}
+                iconDescription="Remove"
```
Comment (scottdickerson) [inline src/components/CardEditor/CardEditForm/CardEditFormItems/ThresholdsFormItem.jsx:241]: i18n

Comment (JoelArmendariz) [reply src/components/CardEditor/CardEditForm/CardEditFormItems/ThresholdsFormItem.jsx:241]: Good catch. Thanks

**Review by scottdickerson — APPROVED** [review]
LGTM

**Review by marcJV — APPROVED** [review]

File: src/components/DashboardEditor/editorUtils.jsx [hunk 2]
```diff
@@ -206,6 +280,44 @@ export const timeRangeToJSON = {
   },
 };
 
+export const validThresholdIcons = [
```
Comment (tay1orjones) [inline src/components/DashboardEditor/editorUtils.jsx:283]: Could these be moved to `src/utils/bundledIcons.js`? The aim there is to have a centralized spot where we import icons for consumption via name through the dashboard/cards

File: src/components/SimpleIconDropdown/SimpleIconDropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+const defaultProps = {
+  icons: validThresholdIcons,
+  light: false,
+  selectedIcon: undefined,
+  testID: undefined,
+  titleText: '',
+};
+
+const SimpleIconDropdown = ({
```
Comment (tay1orjones) [inline src/components/SimpleIconDropdown/SimpleIconDropdown.jsx:40]: Is there a design spec for this component and has it been vetted accepted by the PAL design team? @dianatran18 
https://deploy-preview-1822--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-experimental-simpleicondropdown--default-example
image

To my knowledge this functionality should fall into the [`IconDropdown`](https://deploy-preview-1822--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-experimental-icondropdown--default) pattern/component that's been verified by the PAL design team and is available. 

image

@marcJV  If I recall correctly, it supports icons-only without the text labels. The color mod should also be supported through modifying the Icon passed to the dropdown?

**Comment by tay1orjones** [comment]
> Make sure that all of the functionality updates the JSON properly as defined here: https://www.ibm.com/support/knowledgecenter/en/SSQP8H/iot/dashboard/value_json_ref.html

This isn't going to be feasible in a PR review 😅  If this is a requirement, tests need to be implemented to ensure all Dashboard/card components and helper functions modify and return appropriate json against a defined schema. 

As it stands today we haven't imposed a particular json structure/schema. As much as I'd like to keep the codebase completely agnostic in regards to the json, I realize with each contribution we're getting closer to not being able to remain agnostic and the dashboard components are going to be strongly coupled to the json schema used by Monitor.

@scottdickerson @bryancboyd @davidicus @dianatran18 have we discussed standardizing onto a single dashboard config json format across all of AI Applications?

File: src/components/SimpleIconDropdown/SimpleIconDropdown.jsx (hunk 3 above)
Comment (JoelArmendariz) [reply src/components/SimpleIconDropdown/SimpleIconDropdown.jsx:40]: This was discussed with @CynZhang . Here is the design for the icon/color pickers as it stands for the dashboard editor implementation: 
image
image

We discussed using the 'simple/vertical' dropdowns in this PR for this early implementation, and then updating to these 'horizontal' ones in the next iteration.

**Comment by JoelArmendariz** [comment]
@tay1orjones Apologies, the JSON updating based on our schemas is not a requirement here. I do realize that a lot of the implementations for this are heavily accommodating Monitor-specific requirements.
I think there needs to be a larger discussion regarding a unified JSON schema, but perhaps for now we can move forward with this JSON structure.
FYI @scottdickerson

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**